learning_assistant/api.py

```diff
@@ -307,19 +302,25 @@ def get_or_create_audit_trial(user):
     )
 
 
-def audit_trial_is_expired(audit_trial_data, courserun_key):
-    """
-    Given a user (User), get or create the corresponding LearningAssistantAuditTrial trial object.
+def audit_trial_is_expired(enrollment, audit_trial_data):
     """
-    course_mode = CourseMode.objects.get(course=courserun_key)
+    Given an enrollment and audit_trial_data, return whether the audit trial is expired as a boolean.
+
+    Arguments:
+    * enrollment (CourseEnrollment): the user course enrollment
+    * audit_trial_data (LearningAssistantAuditTrialData): the data related to the audit trial
 
-    upgrade_deadline = course_mode.expiration_datetime()
+    Returns:
+    * audit_trial_is_expired (boolean): whether the audit trial is expired
+    """
+    upgrade_deadline = enrollment.upgrade_deadline
+    today = date.today()
 
     # If the upgrade deadline has passed, return True for expired. Upgrade deadline is an optional attribute of a
```

Comment:
Note: I've changed the way that this is calculated to avoid issues with timezone naive/aware timezones. This now converts all of the datetimes into dates so that we're just calculating the difference in whole days (I believe this is more consistent with the calculations of days remaining in the frontend as well).